Backported from 2.6.15 by Kerin Millar <kerframil@gmail.com>:

From stable-bounces@linux.kernel.org  Sat Jan 14 14:01:19 2006
From: Dmitry Torokhov <dtor_core@ameritech.net>
To: stable@kernel.org
Date: Sat, 14 Jan 2006 16:56:49 -0500
Subject: Input: HID - fix an oops in PID initialization code

Input: HID - fix an oops in PID initialization code

Signed-off-by: Dmitry Torokhov <dtor@mail.ru>
Signed-off-by: Chris Wright <chrisw@sous-sol.org>

--- linux-2.6.14.orig/drivers/usb/input/pid.c	2006-01-18 06:53:14.000000000 +0000
+++ linux-2.6.14/drivers/usb/input/pid.c	2006-01-23 19:05:31.000000000 +0000
@@ -261,7 +261,7 @@
 int hid_pid_init(struct hid_device *hid)
 {
 	struct hid_ff_pid *private;
-	struct hid_input *hidinput = list_entry(&hid->inputs, struct hid_input, list);
+	struct hid_input *hidinput = list_entry(hid->inputs.next, struct hid_input, list);
 
 	private = hid->ff_private = kzalloc(sizeof(struct hid_ff_pid), GFP_KERNEL);
 	if (!private)
